src/core/service.c

```diff
@@ -1252,7 +1252,8 @@ static int service_spawn(
         if (!our_env)
                 return -ENOMEM;
 
-        if ((flags & EXEC_IS_CONTROL) ? s->notify_access == NOTIFY_ALL : s->notify_access != NOTIFY_NONE)
+        if ((flags & EXEC_IS_CONTROL) ? IN_SET(s->notify_access, NOTIFY_ALL, NOTIFY_EXEC)
+                                      : s->notify_access != NOTIFY_NONE)
```

Comment:
i figure this check got complex enough to deserve its function. `service_exec_needs_notify_sockets(Service *s, flags)` or so, with some explanatory comments owuld be great!